controllers/argocd/keycloak.go

```diff
@@ -1248,9 +1362,24 @@ func (r *ReconcileArgoCD) reconcileKeycloakForOpenShift(cr *argoprojv1a1.ArgoCD)
 			log.Info(fmt.Sprintf("Successfully created keycloak realm for ArgoCD %s in namespace %s",
 				cr.Name, cr.Namespace))
 
+			// OAuthClient configuration does not get deleted from previous instances occasionally.
+			// It is safe to delete before updating the OIDC config.
+			err = deleteOAuthClient(cr)
+			if err != nil {
+				return err
+			}
+
 			// Update Realm creation. This will avoid posting of realm configuration on further reconciliations.
+			err = r.Client.Get(context.TODO(), types.NamespacedName{Name: existingDC.Name, Namespace: existingDC.Namespace}, existingDC)
```

Comment:
What we are doing a get here?

Reply:
This was always required. It is better practise to perform Get operation before Update. Any object that we are trying to modify could be updated from the previous copy by user, webhooks or operator for some reason. 

So to avoid the error "Object has ben modified", It is best practise to perform Get before update.